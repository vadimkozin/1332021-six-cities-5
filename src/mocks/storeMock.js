import configureStore from 'redux-mock-store';
import {SortingType, CITIES_NAMES, StatusType, AuthorizationStatus} from '../const';
import {offerAdapter, commentAdapter} from '../store/adapters';

import hotelJson from './json/hotel';
import hotelsNearbyJson from './json/hotels-nearby';
import offersJson from './json/offers';
import commentsJson from './json/comments';

const hotel = offerAdapter.adaptToClientOffer(hotelJson);
const hotelsNearby = offerAdapter.adaptToClient(hotelsNearbyJson);
const offers = offerAdapter.adaptToClient(offersJson);
const comments = commentAdapter.adaptToClient(commentsJson);

const initCity = CITIES_NAMES[0];

const Data = {
  SORT: `sort`,
  CITY: `city`,
  OFFERS: `offersGist`,
  HOTEL: `hotelGist`,
  HOTELS_NEARBY: `hotelsNearbyGist`,
  COMMENTS: `commentsGist`,
};

const basicState = {
  data: null,
  status: StatusType.IDLE,
  error: ``,
};

const makeInit = (basic = basicState) => (data = {}) => Object.assign({}, basic, {data});

const init = makeInit();

const storeConfig = configureStore();

const storeMock = storeConfig({
  data: {
    cities: CITIES_NAMES,
  },
  process: {
    [Data.CITY]: initCity,
    [Data.SORT]: SortingType.POPULAR,
    [Data.OFFERS]: init(offers),
    [Data.HOTEL]: init(hotel),
    [Data.HOTELS_NEARBY]: init(hotelsNearby),
    [Data.COMMENTS]: init(comments),
  },
  user: {
    id: 1,
    name: `qwerty`,
    email: `qwerty@mail.ru`,
    isPro: false,
    avatarUrl: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
});

export {storeMock};
