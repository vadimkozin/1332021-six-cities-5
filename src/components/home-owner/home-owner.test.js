import React from 'react';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import {render} from 'enzyme';
import storeMock, {offers} from '../../mocks/storeMock';
import HomeOwner from './home-owner';

const offer = offers[0];

describe(`HomeOwner`, () => {
  it(`should render HomeOwner component`, () => {
    const component = render(
        <Provider store={storeMock}>
          <MemoryRouter>
            <HomeOwner owner={offer.owner} description={offer.description} />
          </MemoryRouter>
        </Provider>
    );

    expect(component).toMatchSnapshot();
  });
});
