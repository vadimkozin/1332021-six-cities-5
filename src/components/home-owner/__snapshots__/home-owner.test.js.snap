// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeOwner should render HomeOwner component 1`] = `
<div
  class="property__host"
>
  <h2
    class="property__host-title"
  >
    Meet the host
  </h2>
  <div
    class="property__host-user user"
  >
    <div
      class="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
    >
      <img
        alt="Host avatar"
        class="property__avatar user__avatar"
        height="74"
        src="img/avatar-angelina.jpg"
        width="74"
      />
    </div>
    <span
      class="property__user-name"
    >
      Angelina
    </span>
    <span
      class="property__user-status"
    >
      Pro
    </span>
  </div>
  <div
    class="property__description"
  >
    <p
      class="property__text"
    >
      Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
    </p>
  </div>
</div>
`;
